### DISCUSSION TITLE
Horizontal Axis label positioning

### DISCUSSION THREAD

**Discussion by virtualpathum** [patrykandpatrick/vico#637] [Questions]
Hi, I need to position the x-axis label exactly as follows.  

![image](https://github.com/patrykandpatrick/vico/assets/3455658/9915dec4-0978-41b6-9851-1b83fff10c90)

I tried playing with the following configs but couldn't make it work exactly as I wanted. 

```
private val bottomAxisItemPlacerFT = AxisItemPlacer.Horizontal.default(
    spacing = BOTTOM_AXIS_ITEM_SPACING,
    offset = BOTTOM_AXIS_ITEM_OFFSET,
    shiftExtremeTicks = true,
    addExtremeLabelPadding = false)

fun getBottomAxisValueFormatter(xAxisData: List<String>): AxisValueFormatter<AxisPosition.Horizontal.Bottom> {
    return AxisValueFormatter { x, _, _ ->
        xAxisData[x.toInt() % xAxisData.size]+"        " // added space to move the label to the left. 
    }
}

rememberBottomAxis label padding/ text alignment doesn't help
```

Any suggestions?

I am using Jetpack Compose with VICO v2.0.0-alpha.11

**Comment by patrickmichalik** [comment]
Hello! You can do this by creating a custom `AxisItemPlacer.Horizontal` implementation. Please see the functions’ documentation for information on what they should return. Since `ColumnCartesianLayer` columns are centered relative to their _x_ values, each label will be shifted left in relation to its _x_ value. To ensure that the label text is correct, you can correct for this shift at the formatting stage.

Another option is to create a `HorizontalAxis` subclass and make the required overrides. This is possible in Vico 2.0.0 Alpha 15 and later. Such an approach is a bit more low-level but gives you the greatest flexibility. That said, here, `AxisItemPlacer.Horizontal` should suffice.

If you have any questions, let me know.

**Reply by virtualpathum** [reply]
@patrickmichalik Thanks for the suggestion. At the moment I am using 2.0.0-alpha.11. it looks like there have been a lot of changes in alpha 17. I am not sure I should change the version at this point. Anyway, for now, my solution seems working and probably with the next upgrade I will do this. Thanks again

**Reply by patrickmichalik** [reply]
You’re welcome! Sounds good. I’m glad you’ve found a satisfactory solution.
